src/GraphQL.Client/GraphQLHttpClient.cs

```diff
@@ -63,7 +63,8 @@ public GraphQLHttpClient(GraphQLHttpClientOptions options, IGraphQLWebsocketJson
             if (!HttpClient.DefaultRequestHeaders.UserAgent.Any())
                 HttpClient.DefaultRequestHeaders.UserAgent.Add(new ProductInfoHeaderValue(GetType().Assembly.GetName().Name, GetType().Assembly.GetName().Version.ToString()));
 
-            _graphQlHttpWebSocket = new GraphQLHttpWebSocket(GetWebSocketUri(), this);
+            if (options.UseWebSocketForQueriesAndMutations)
```

Comment:
Subscriptions must always use the websocket connection (obviously they don't work with regular HTTP requests). 

You need to make sure that an instance of `GraphQLWebSocket` is created when a subscription is created, otherwise you will receive a null reference exception on creating a subscription.